lib/common/validation.js

```diff
@@ -456,6 +460,47 @@ var validators = {
 
     zfs_io_priority: createValidateNumberFn('zfs_io_priority'),
 
+    zfs_snapshot_limit: function (params) {
+        var errs = [];
+
+        /*
+         * This parameter nominally takes an integer, but it must be
+         * also able to take null as a way to disable limits on
+         * snapshot.  You can completely restrict snapshots with a
+         * value of 0, so 0 can't be used for disabling.
+         *
+         * Worse, is that vmadm(1M) uses the empty string ('') for
+         * disabling limits, so vmadm(1M)/the-PI must be updated to
+         * deal with translating VMAPI's null.
+         *
+         * NOTE that VMAPI doesn't accept empty strings.
+         *
+         * This constitutes a Flag Day for PIs that wish to use this VMAPI
+         * function.  Any pre-flag-day PI can use this feature, but a null
+         * will not cancel snapshot limit enforcement, but will lower it to
+         * 0 (i.e. NO SNAPSHOTS ALLOWED).
```

Comment:
I don't think we need this vmadm flag-day comment - vmapi shouldn't care about the internals of vmadm.

Reply:
Users of VMAPI will be bitten by violations of least-surprise on older PIs, but the source is probably not the best place to document that.